test/functional/rpc_blockchain.py

```diff
@@ -429,6 +430,17 @@ def _test_getdifficulty(self):
     def _test_getnetworkhashps(self):
         self.log.info("Test getnetworkhashps")
         hashes_per_second = self.nodes[0].getnetworkhashps()
+        assert_raises_rpc_error(
+            -3,
+            textwrap.dedent("""
+            Wrong type passed:
+            {
+                "Position 1 (nblocks)": "JSON value of type string is not of expected type number",
+                "Position 2 (height)": "JSON value of type array is not of expected type number"
+            }
+            """).strip(),
+            lambda: self.nodes[0].getnetworkhashps("a", []),
+        )
```

Comment:
I don't think this is an appropriate place for this test?

E.g. renaming `rpc_named_arguments.py` -> `rpc_arguments.py` and adding it there seems more logical to me.

Reply:
Why not? Every other RPC that has missing-arg or wrong-type checks is also called in the rpc-specific test file?

Examples:

```
git grep 'is not of expected type' test
```

For example in this file:

```
test/functional/rpc_blockchain.py:        assert_raises_rpc_error(-3, "JSON value of type string is not of expected type number", self.nodes[0].getchaintxstats, '')
test/functional/rpc_blockchain.py:        assert_raises_rpc_error(-3, "JSON value of type number is not of expected type string", self.nodes[0].getchaintxstats, blockhash=0)
test/functional/rpc_blockchain.py:        assert_raises_rpc_error(-3, "JSON value of type string is not of expected type number", node.getblock, blockhash, "2")

Reply:
Oh yes, I'm not arguing against checking for an RPC's specific argument types within the RPC-specific test file in general, that can help with regressions etc.

Since this PR is about type checking in general (and not about `getnetworkhashps`), I think it'd be good to have a test that checks the correct error message is raised - and that should be a test dedicated to just that? If for some unrelated reason in the future `_test_getnetworkhashps()` gets deleted because the endpoint is deprecated, we'd unintentionally remove test coverage on this PR too.

Reply:
Ok, will add more tests in the follow-up